src/Search.f90

```diff
@@ -11,95 +11,87 @@
 MODULE Search
 USE IntrType
 USE Strings
-IMPLICIT NONE
 
+IMPLICIT NONE
 PRIVATE
 
 PUBLIC :: upperBound
 PUBLIC :: lowerBound
 PUBLIC :: find
 PUBLIC :: binarySearch
 
-!> @brief Return the index of the first element of a list that compares
-!> greater than the passed value
-!> @param list the list to search. The list should be sorted, otherwise
-!> undefined behavior.
-!> @param val the value to find
-!> @param tol the comparison tolerance to use for floating-point
-!> implementations. Optional
-!> @returns the index to the first element of the list that compares greater
-!> than the passed search value
+!> @brief Interface for getting "upper bounds"
 !>
-!> If the value is greater than or equal to the largest element of the list,
-!> returns SIZE + 1.
 INTERFACE upperBound
```

Comment:
Didn't we talk about renaming these functions to something better?  `FindFirstGreater` and `FindFirstGreaterOrEqual` or some such thing?  (I'm VERY open to better names as long as they're accurate/descriptive, but these ones are very unclear).

Reply:
Done.